sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,384 @@
+    template <class T> complex<T> log10(const complex<T> &);
+    /// Compute the square root of complex number x.
+    template <class T> complex<T> sqrt(const complex<T> &);
+    /// Compute the base-e exponent of complex number x.
+    template <class T> complex<T> exp(const complex<T> &);
+
+    /// Compute complex number z raised to the power of complex number y.
+    template <class T> complex<T> pow(const complex<T> &, const complex<T> &);
+    /// Compute complex number z raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const complex<U> &);
+    /// Compute complex number z raised to the power of real number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const U &);
+    /// Compute real number x raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const T &, const complex<U> &);
+
+    /// Compute the inverse hyperbolic sine of complex number x.
+    template <class T> complex<T> asinh(const complex<T> &);
+    /// Compute the inverse hyperbolic cosine of complex number x.
+    template <class T> complex<T> acosh(const complex<T> &);
+    /// Compute the inverse hyperbolic tangent of complex number x.
+    template <class T> complex<T> atanh(const complex<T> &);
+    /// Compute the hyperbolic sine of complex number x.
+    template <class T> complex<T> sinh(const complex<T> &);
+    /// Compute the hyperbolic cosine of complex number x.
+    template <class T> complex<T> cosh(const complex<T> &);
+    /// Compute the hyperbolic tangent of complex number x.
+    template <class T> complex<T> tanh(const complex<T> &);
+    /// Compute the inverse sine of complex number x.
+    template <class T> complex<T> asin(const complex<T> &);
+    /// Compute the inverse cosine of complex number x.
+    template <class T> complex<T> acos(const complex<T> &);
+    /// Compute the inverse tangent of complex number x.
+    template <class T> complex<T> atan(const complex<T> &);
+    /// Compute the sine of complex number x.
+    template <class T> complex<T> sin(const complex<T> &);
+    /// Compute the cosine of complex number x.
+    template <class T> complex<T> cos(const complex<T> &);
+    // Compute the tangent of complex number x.
+    template <class T> complex<T> tan(const complex<T> &);
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+== Suffixes for complex number literals
+
+This proposal describes literal suffixes for constructing complex number
+literals.
+The suffixes `i` and `if` create complex numbers of the types `complex<double>`
+and `complex<float>` respectively, with their imaginary part denoted by the
+given literal number and the real part being zero.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+namespace literals {
+namespace complex_literals {
+constexpr complex<double> operator""i(long double x);
```

Comment:
This cannot be used in device code because `long double` is a forbidden feature in device code.  I suppose you could add it for use only in host code, but I'm not sure how valuable it is to have support only in host code for these literals.  Another option is to use raw literal operators, but this is admittedly a lot less convenient.

There is a discussion on this topic in KhronosGroup/SYCL-Docs#379 (search for "user defined floating point literal" in that issue).  The solution proposed there would require either a change to the core SYCL spec or an extension to DPC++.  DPC++ does not currently implement this support.  In addition, it requires use of `consteval`, which is a C++20 feature.

Reply:
Hi @abagusetty, you've recently changed the sycl::complex's proposal extension.

This PR is moving the extension to the experimental folder.
I've pulled your modifications to this PR recently and received feedback regarding your section.

I just wanted to ping you because you might be best suited to explain the reason for introducing these literals and answering these comments.

Reply:
The motivation behind supporting the float literals was from the needs of device-side. As Greg was pointing, not much valuable from the host-side.
Since `long double` is not a fundamental data type: we can change the arguments to `double` as a short-term solution. 

I presume, the expt. spec can be updated in the future to adopt user-defined literals and when the spec bumps the feature support to adopt C++20 standards.

Reply:
> we can change the arguments to `double` as a short-term solution

Does this work?  You should run a test and find out.  I thought the C++ standard required the argument signature to be either `long double` (for the "cooked" form") or `char *` (for the "raw" form).  My guess is that the compiler will not parse the user-defined literal if the function signature is `double`, but you should check.